doc/source/rllib/getting-started.rst

```diff
@@ -0,0 +1,449 @@
+    │ PPO_Pendulum-v1_b5c41_00002   RUNNING    0.0001       20            60.0421             -960.293                   80000 │
+    ╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
+
+``Tuner.fit()`` returns an ``ResultGrid`` object that allows for a detailed analysis of the
+training process and for retrieving the :ref:`checkpoints <rllib-checkpoints-docs>` of the trained
+algorithms and their models:
+
+.. testcode::
+    # Get the best result based on a particular metric.
+    best_result = results.get_best_result(
+        metric="env_runners/episode_return_mean", mode="max"
+    )
+
+    # Get the best checkpoint corresponding to the best result
+    # from the preceding experiment.
+    best_checkpoint = best_result.checkpoint
+
+
+Deploy a trained model for inference
+++++++++++++++++++++++++++++++++++++
+
+After training, you might want to deploy your models into a new environment, for example
+to run inference in production. For this purpose, you can use the checkpoint directory created
+in the preceding example. To read more about checkpoints, model deployments, and restoring algorithm state,
+see this :ref:`page on checkpointing <rllib-checkpoints-docs>` here.
+
+Here is how you would create a new model instance from the checkpoint and run inference through
+a single episode of your RL environment. Note in particular the use of the
+:py:meth:`~ray.rllib.utils.checkpoints.Checkpointable.from_checkpoint` method to create
+the model and the
+:py:meth:`~ray.rllib.core.rl_module.rl_module.RLModule.forward_inference`
+method to compute actions:
+
+
+.. testcode::
+
+    from pathlib import Path
+    import gymnasium as gym
+    import numpy as np
+    import torch
+    from ray.rllib.core.rl_module import RLModule
+
+    # Create only the neural network (RLModule) from our algorithm checkpoint.
+    rl_module = RLModule.from_checkpoint(
+        Path(best_checkpoint.path) / "learner_group" / "learner" / "rl_module"
+    )["default_policy"]
+
+    # Create the RL environment to test against (same as was used for training earlier).
+    env = gym.make("Pendulum-v1", render_mode="human")
+
+    episode_return = 0.0
+    done = False
+
+    # Reset the env to get the initial observation.
+    obs, info = env.reset()
+
+    while not done:
+        # Render the env.
+        env.render()
+
+        # Compute the next action from a batch (B=1) of observations.
+        obs_batch = torch.from_numpy(obs).unsqueeze(0)  # add batch B=1 dimension
+        model_outputs = rl_module.forward_inference({"obs": obs_batch})
+        # Extract the logits from the output and dissolve batch again.
+        action_logits = model_outputs["action_dist_inputs"][0]
+        # PPO's default RLModule produces action logits (from which
+        # you have to sample an action or use the max-likelihood one).
+        action = numpy.argmax(action_logits.numpy())
```

Comment:
I admit, this is an easy setup and could support understanding how exactly the "optimal" action is computed in inference. However the way how we do it in evaluation is to build the corresponding action distribution - here `DiagGaussian` and convert it to a deterministic one - using `to_deterministic` - doing basically what we show above explicitly. I am unsure, if we might show this here or the simplisitic form that is shown here already.

Reply:
🤔